### ISSUE TITLE
Reconsider save and publishing flow for keyboard users

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#30634]
## Description

After making a change to the site editor and clicking tab, a prompt shows to open the saving panel but the focus doesn't go to the prompt. This came up during a hallway hangout for the FSE outreach program. Props to @carolinan for finding it!

## Step-by-step reproduction instructions

1. Create/use a site with FSE capabilities.
2. Open the site editor. 
3. Make a change and hit tab.
4. Notice a saving prompt appears but the keyboard focus doesn't shift to it. 

## Expected behaviour

I expect upon tabbing for the focus to go to the saving prompt to allow me to open the save panel and save. 

## Actual behaviour

The focus doesn't go to the saving prompt. 

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/26996883/114080830-0fdc0e80-9869-11eb-8538-c00e43d8f51b.mov

## WordPress information
- WordPress version: 5.7
- Gutenberg version: 10.3.2
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1 Blocks

## Device information
- Device: Desktop
- Operating system: MacOS
- Browser: Chrome v 89

**Comment by carolinan** [comment]
I am not sure that this prompt is even needed:

1. It assumes that the user wants to save as soon as the edited block is no longer in focus. They may want to create or edit more than one block before saving.
2. It assumes that the user has not already closed the save panel on purpose.

**Comment by annezazu** [comment]
Great points. I'll flag this for the accessibility team to see if we can get more thoughts here!

**Comment by joedolson** [comment]
I just tested this, and in my test the keyboard focus *did* move to the save prompt, although the visible focus was not obvious to me - the focus state of .is-secondary buttons doesn't really look focused unless you *also* saw the button prior to receiving focus. 

I think that a prompt that indicates that there are dirty posts that need to be saved is important, but I think that it should be persistent - perhaps located in the toolbar - so that it can always be found when needed, rather than appearing and disappearing unpredictably & forcing a change of context.

The focus move is not helpful - like @carolinan describes, it will interrupt the editing process by moving keyboard users out of the editing flow after every single block edit, which makes for significant problems. It would be a better flow if there was a persistent indicator that a save was needed, and a user could predictably find it in an area that is already populated with other important controls.

**Comment by carolinan** [comment]
I just noticed that this prompt is also in the post editor. I had not seen this before. I'd like to track back to where this was added to see if there were any related discussions.

![An "Open publish panel" prompt is visible in the bottom right corner of the post editor screen](https://user-images.githubusercontent.com/7422055/114251500-8845eb00-99a1-11eb-942d-eccc4b1e7ec5.png)



As soon as something has changed, the save button in the top toolbar is enabled, is that indicator not enough?
Are there other indicators in the post editor that are not used in the site editor?

**Comment by carolinan** [comment]
It looks like this can be traced back all the way to 2018 https://github.com/WordPress/gutenberg/pull/7552
I really had no idea this prompt existed.

**Comment by joedolson** [comment]
Huh - I'd never seen that before, either. In the post editor, if you have the sidebar open, it seems to only appear if you tab all the way through the panel; seems pretty messy. 

This probably needs some thinking through. When navigating through the block panel, you get a floating skip link panel that shifts focus to the selected block, then this 'open publish panel' block - the two behave similarly (only visible on focus), but look completely different and appear in different locations. 

I feel like the whole interface would be clearer if both of these options were just *visible*, rather than this unpredictable behavior.

**Comment by carolinan** [comment]
Here is a gif describing the two links in the post editor that Joe mentions above.
Note also that the skip link to the selected block covers the "Advanced" settings.

![Tabbing through the block settings panel, a skip link to the selected block is floating and blocking the view of the last setting in the panel. Tabbing again closes this skip link and focus is on a previously hidden panel with a link to the save panel.](https://user-images.githubusercontent.com/7422055/114702236-19abb900-9d24-11eb-9555-40bb11df04f8.gif)

**Comment by carolinan** [comment]
While testing https://github.com/WordPress/gutenberg/pull/30873 I confirmed that the open save panel prompt does receive focus. I agree that the flow is still disruptive.

**Comment by annezazu** [comment]
@david-szabo97 wanted to be sure you saw this thread and feedback in case you might be able to follow up here!

**Comment by david-szabo97** [comment]
Focus works for me as it was mentioned throughout the thread.

> As soon as something has changed, the save button in the top toolbar is enabled, is that indicator not enough?

Same question. It looks like that should be enough and this prompt seems unnecessary. It's really confusing when the sidebar is open 😕 On the other hand, I'm not really experienced in this area. For landmark navigation it makes sense, but we have the Save button in the header toolbar so we can just navigate there and trigger Save.
